### PR TITLE
Remove `clearable` from color and time pickers

### PR DESCRIPTION
**Pull request by andrearutrecht**
The `clearable` prop is removed from all pickers in `admin/admin-date-time` and from `ColorPicker`.  All Pickers are now `clearable` by default and only not `clearable` when `required` is set.

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-670
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots of Time and Date Pickers in Storybook to show default `clearable` behaviour:</summary>

Screenshot 2024-04-19 at 13 51 29

</details>

<details><summary>Screenshots of Time and Date Pickers in Storybook to show `required` behaviour:</summary>
<p>

Screenshot 2024-04-23 at 11 17 44
Screenshot 2024-04-23 at 11 17 52
Screenshot 2024-04-23 at 11 17 58
Screenshot 2024-04-23 at 11 18 03
Screenshot 2024-04-23 at 11 18 09

</p>
</details> 

<details><summary>Screenshots of `ColorPicker` stories:</summary>
<p>

Screenshot 2024-04-22 at 09 47 18
Screenshot 2024-04-22 at 09 47 30
Screenshot 2024-04-22 at 09 47 38

</p>
</details>

### CODE REVIEW

**Comment by johnnyomair** [comment]
@andrearutrecht is this ready for review?

**Comment by andrearutrecht** [comment]
> @andrearutrecht is this ready for review?

@johnnyomair No, it is still set to draft

**Comment by johnnyomair** [comment]
> > @andrearutrecht is this ready for review?
> 
> @johnnyomair No, it is still set to draft

Then the review requests should be removed.

Draft + review request = "It can't be merged like this, but I want your opinion on it anyway"

Draft + no review request = "I'm still working on it"

**Comment by andrearutrecht** [comment]
Oh, thank you for that info! I did not know.

File: storybook/src/docs/form/components/DateAndTimePickers/stories/DateTimePicker.stories.tsx [hunk 1]
```diff
@@ -20,8 +20,8 @@ storiesOf("stories/form/components/Date & Time Pickers/Date-Time Picker", module
                     </FieldContainer>
                 </Grid>
                 <Grid item xs={12} md={6}>
-                    <FieldContainer label="Clearable" fullWidth>
-                        <DateTimePicker value={dateTimeTwo} onChange={setDateTimeTwo} clearable />
+                    <FieldContainer label="Required" fullWidth required={true}>
+                        <DateTimePicker value={dateTimeTwo} onChange={setDateTimeTwo} required={true} />
```
Comment (jamesricky) [inline storybook/src/docs/form/components/DateAndTimePickers/stories/DateTimePicker.stories.tsx:24]:
```suggestion
                    <FieldContainer label="Required" fullWidth required>
                        <DateTimePicker value={dateTimeTwo} onChange={setDateTimeTwo} required />
```

File: .changeset/light-ligers-wonder.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+"@comet/admin-date-time": major
+---
+
+Remove `clearable` prop from `DateRangePicker`, `DateTimePicker`, `TimePicker` and `TimeRangePicker`
+
+Add `required` prop to `DateRangePicker`, `DateTimePicker`, `TimePicker` and `TimeRangePicker`
```
Comment (jamesricky) [inline .changeset/light-ligers-wonder.md:7]: Maybe both should be mentioned in the title 🤔 

```suggestion
Remove the `clearable` prop and add a `required` prop to `DateRangePicker`, `DateTimePicker`, `TimePicker` and `TimeRangePicker`
```

Comment (jamesricky) [reply .changeset/light-ligers-wonder.md]: @andrearutrecht please do the same for the color-picker changeset.

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**